battle.go

```diff
@@ -140,78 +141,93 @@ func (b *Battle) SimulateRound() ([]Transaction, bool) {
 		case ItemTurn:
 			receiver := b.getPokemon(t.Target.party, t.Target.partySlot)
 			move := receiver.Moves[t.Move]
-			queue = append(queue, ItemTransaction{
+			b.QueueTransaction(ItemTransaction{
 				Target: receiver,
 				Item:   t.Item,
 				Move:   move,
 			})
-			queue = append(queue, receiver.UseItem(t.Item)...)
+			b.QueueTransaction(receiver.UseItem(t.Item)...)
 		default:
 			log.Panicf("Unknown turn of type %v", t)
 		}
-		// process transations for this turn
-		for len(queue) > 0 {
-			next := queue[0]
-			queue = queue[1:]
-			switch t := next.(type) {
-			case DamageTransaction:
-				if t.Target.CurrentHP >= t.Damage {
-					t.Target.CurrentHP -= t.Damage
-				} else {
-					// prevent underflow
-					t.Target.CurrentHP = 0
-				}
-				if t.Target.CurrentHP == 0 {
-					// pokemon has fainted
-					queue = append(queue, FaintTransaction{
-						Target:          t.Target,
+
+		b.ProcessQueue()
+	}
+
+	if len(b.tQueue) > 0 {
+		log.Panic("FATAL: There are still unprocessed transactions at the end of the round.")
+	}
+	transactions := b.tProcessed
+	b.tProcessed = []Transaction{}
+	return transactions, b.State == BATTLE_END
+}
+
+// Add Transactions to the queue.
+func (b *Battle) QueueTransaction(t ...Transaction) {
```

Comment:
I really like this function, its basic but I def thinks it helps improve readability of the battle logic